tests/playbooks/k8s_auth.yml

```diff
@@ -13,34 +18,22 @@
 
     - name: Log in (obtain auth token)
       k8s_auth:
-        host: https://192.168.42.55:8443
-#       ssl_ca_cert: ca.pem
-        verify_ssl: no
         username: developer
         password: "{{ k8s_developer_password }}"
       register: k8s_auth_results
 
-    - name: Preserve auth info as both a fact and a yaml anchor for easy access later
-      # Both the fact and the anchor are called 'k8s_auth_params'
-      set_fact:
-        k8s_auth_params: &k8s_auth_params
-          host: "{{ k8s_auth_results.k8s_auth.host }}"
-          ssl_ca_cert: "{{ k8s_auth_results.k8s_auth.ssl_ca_cert }}"
-          verify_ssl: "{{ k8s_auth_results.k8s_auth.verify_ssl }}"
-          api_key: "{{ k8s_auth_results.k8s_auth.api_key }}"
-
     # Previous task generated I(k8s_auth) fact, which you can then use
     # in k8s modules like this:
     - name: Get a list of all pods from any namespace
       k8s_facts:
-        <<: *k8s_auth_params
+        api_key: "{{ k8s_auth_results.k8s_auth.api_key }}"
```

Comment:
Is it possible to have this as part of defaults as well?

Reply:
Only after I write a `set_module_default` module (analogous to `set_fact`), which I may or may not have already investigated. :)

For the time being: no. But it will be glorious once it's possible. :)